frontend/include/chpl/parsing/parsing-queries.h

```diff
@@ -616,6 +616,15 @@ ID idToContainingMultiDeclId(Context* context, ID id);
 bool idContainsFieldWithName(Context* context, ID typeDeclId,
                              UniqueString fieldName);
 
+/**
+  Given an AST node for a (multi-)declaration, find for a Variable
+  node that declares a field of the given name. Returns whether the field
+  was found, and, if it was, sets outFieldId to the ID of the Variable.
+ */
+bool findFieldIdInDeclaration(const uast::AstNode* varDecl,
+                              UniqueString fieldName,
+                              ID& outFieldId);
```

Comment:
You could consider having this return an empty ID if nothing is found.

Reply:
because this is a recursive I think it might perform several assignments, but if that's our bottleneck for performance, we can revisit it later. For now, I'll have it return empty.